File: languages/php/oop/01_pessoa.php
<?php
/*
Objetivo: introduzir classes e objetos em PHP.
Como executar: php .\languages\php\oop\01_pessoa.php
Conceitos treinados: classe, construtor, métodos, encapsulamento simples.
Próximo exercício: herança e composição.
*/

class Pessoa {
    private string $nome;
    private string $area;

    public function __construct(string $nome, string $area) {
        $this->nome = $nome;
        $this->area = $area;
    }

    public function apresentar(): string {
        return "Olá, meu nome é {$this->nome} e estudo {$this->area}.";
    }
}

$usuario = new Pessoa("Juniormar", "programação");
echo $usuario->apresentar() . PHP_EOL;

File: languages/java/oop/01_Pessoa.java
/*
Objetivo: introduzir classes e objetos em Java.
Como executar:
  cd .\languages\java\oop
  javac 01_Pessoa.java
  java PessoaTeste
Conceitos treinados: classe, atributo, construtor, método.
Próximo exercício: herança.
*/

class Pessoa {
    private String nome;
    private String area;

    public Pessoa(String nome, String area) {
        this.nome = nome;
        this.area = area;
    }

    public String apresentar() {
        return "Olá, meu nome é " + nome + " e estudo " + area + ".";
    }
}

class PessoaTeste {
    public static void main(String[] args) {
        Pessoa usuario = new Pessoa("Juniormar", "programação");
        System.out.println(usuario.apresentar());
    }
}
